snuba/utils/kafka/consumers/abstract.py

```diff
@@ -0,0 +1,34 @@
+from abc import ABC, abstractmethod
+from typing import Callable, Generic, NewType, Optional, Sequence, Tuple, TypeVar
+
+
+TMessage = TypeVar("TMessage")
+
+Topic = NewType("Topic", str)
+Partition = NewType("Partition", int)
+Offset = NewType("Offset", int)
+
+
+class Consumer(Generic[TMessage], ABC):
+    @abstractmethod
+    def subscribe(
+        self,
+        topics: Sequence[str],
+        on_assign: Optional[Callable[[Sequence[Tuple[Topic, Partition]]], None]] = None,
+        on_revoke: Optional[Callable[[Sequence[Tuple[Topic, Partition]]], None]] = None,
+    ) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    def poll(self, timeout: Optional[float] = None) -> Optional[TMessage]:
+        raise NotImplementedError
+
+    @abstractmethod
+    def commit(
+        self, asynchronous: bool = True
```

Comment:
The confluent consumer allows the caller to pass an offset or a message, was this dropped intentionally ? (StrictConsumer relies on that).

Reply:
It's dropped intentionally, sort of — I left it out because nothing that subclasses it is using it directly and am trying to minimize the amount of work (and testing) that needs to be done with the Confluent wrapper.

I am fine with adding it in the future when `StrictConsumer` acts as a subclass of this (or wraps this and uses it internally.) I'd prefer to do that later, though.

Reply:
As long as there is no strong reason not to have the parameter, I am ok waiting when we will support StrictConsumer.